skimage/transform/_geometric.py

```diff
@@ -1447,7 +1448,8 @@ def estimate(self, src, dst, order=2):
         A[:rows, -1] = xd
         A[rows:, -1] = yd
 
-        _, _, V = np.linalg.svd(A)
+        # scipy sparse svd uses less memory
+        _, _, V = svds(A, return_singular_vectors = True)
```

Comment:
```suggestion

        if A.shape[1] <= A.shape[0]:
            # scipy sparse svd uses less memory
            _, _, V = svds(A, k=1, which='SM', return_singular_vectors='vh')
        else:
            # use full SVD in the overdetermined case
            _, _, V = np.linalg.svd(A, full_matrices=True)
```

I played around with this and the `svds` call as above should return only the smallest singular value. This failed two test cases, though. In the overdetermined case, `V[-1]` is not the same `V[-1]` as when using a reduced SVD because `V.shape[0]` is greater than the number of singular values in A. If I restrict to using `svds` only in the non-overdetermined cases as above, then I only saw one precision-related failure in `test_polynomial_default_order`. It passes if I set `decimal=5` in the call `assert_almost_equal`.

I'm not sure if there is a better solution (I am not a linear algebra guru), but maybe this helps? I would be a little concerned that we don't have a ton of test cases, though, so I can't be super confident on the extent that precision changes might be an issue in real-world use cases.

Reply:
clarification: when always using the `svds` path two test cases fail. With the if/else as suggested above there is only the one precision-related failure that can be resolved by changing to `decimal=5` in the test.